src/main/java/io/github/apace100/origins/mixin/LoginMixin.java

```diff
@@ -71,7 +72,7 @@ private void openOriginsGui(ClientConnection connection, ServerPlayerEntity play
 		ServerSidePacketRegistry.INSTANCE.sendToPlayer(player, ModPackets.LAYER_LIST, originLayerData);
 
 		List<ServerPlayerEntity> playerList = getPlayerList();
-		playerList.forEach(spe -> ModComponents.ORIGIN.get(spe).syncWith(player));
+		playerList.forEach(spe -> ModComponents.ORIGIN.syncWith(spe, ComponentProvider.fromEntity(player)));
```

Comment:
Yes, this is necessary - otherwise newly joined clients will not receive the component's information of others, e.g. previously joined Phantoms will not appear translucent, or even invisible when in their Phantom state, to the new client.

Reply:
I mean, specifically synchronizing with every client on the server. I'm noting in the next line you are already synchronizing with every player watching the newly joined client using the regular sync method, so there seems to be a redundancy there.

Reply:
Note that it does not, like you said, synchronize with every client on the server - it synchronizes _the data of every client's component_ with the client who newly joins.

But I think I see now, what you mean - you're asking whether the client that is currently joining is already in the list and therefore doesn't need their component synced in the line directly after this `forEach`? Because the review showed the `forEach`, but not the line after, I thought you were referring to the synchronization of all client's component data.

Thanks for pointing it out! I checked and it seems by the end of the `onPlayerConnected` method (where this mixin is situated), the player is already part of the player list. I'm therefore going to remove the single `OriginComponent#sync` call and keep just the loop.

Reply:
> Note that it does not, like you said, synchronize with every client on the server - it synchronizes the data of every client's component with the client who newly joins

Oh I see, this is indeed different. I'm still surprised this is needed though. May be a bug in CCA ?